src/api/raw.rs

```diff
@@ -196,12 +196,86 @@ pub struct ListParams {
 #[derive(Default, Clone)]
 pub struct PostParams {
     pub dry_run: bool,
 }
 
+/// Common query parameters for patch calls
+#[derive(Default, Clone)]
+pub struct PatchParams {
+    pub dry_run: bool,
+    /// Strategy which will be used. Defaults to `PatchStrategy::Strategic`
+    pub patch_strategy: PatchStrategy,
+    /// force Apply requests. Applicable only to `PatchStrategy::Apply`
+    pub force: bool,
+    /// fieldManager is a name of the actor that is making changes. Required for `PatchStrategy::Apply`
+    /// optional for everything else
+    pub field_manager: Option<String>
+}
+
+impl PatchParams {
+    fn validate(&self) -> Result<()> {
+        if let Some(field_manager) = &self.field_manager {
+            // that's not complete, go reference code does list all Unicode printable characters as valid, 
+            // this validation is a bit stricter 
+            if field_manager.len() > 128 || !field_manager.chars().all(|f| f.is_alphanumeric() || f == ' ') {
+            return Err(ErrorKind::RequestValidation("Failed to validate PatchParameters::field_manager!".to_owned()).into())
+            }
+        }
+        
+        if self.patch_strategy != PatchStrategy::Apply && self.force {
+             // if not force, all other fields are valid for all types of patch requests
+            Err(ErrorKind::RequestValidation("Force is applicable only for Apply strategy!".to_owned()).into())
+        } else {
+            Ok(())
+        }
+    }
+
+    fn populate_qp(&self, qp: &mut url::form_urlencoded::Serializer<String>) {
+        if self.dry_run {
+            qp.append_pair("dryRun", "true");
+        }
+        if self.force {
+            qp.append_pair("force", "true");
+        }
+        if self.field_manager.is_some() {
+            qp.append_pair("fieldManager", self.field_manager.as_ref().unwrap());
+        }
+    }
+}
+
+
+/// For patch different patch types are supported. See https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment
+/// Apply strategy is kinda special
+#[derive(Clone, PartialEq)]
+pub enum PatchStrategy {
+    Apply,
+    JSON,
+    Merge,
+    Strategic
+}
+
+impl std::fmt::Display for PatchStrategy {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let content_type = match &self {
+            PatchStrategy::Apply => "application/apply-patch+yaml",
+            PatchStrategy::JSON => "application/json-patch+json",
+            PatchStrategy::Merge => "application/merge-patch+json",
+            PatchStrategy::Strategic => "application/strategic-merge-patch+json"
+        };
+        f.write_str(content_type)
+    }
+}
+
+// Kubectl defaults to Strategic strategy, but doing so will break existing consumers
+// so, currently we still default to Merge it may change in future versions
+// Strategic merge doesn't work with CRD types https://github.com/kubernetes/kubernetes/issues/52772
```

Comment:
I think it's okay to make a breaking change here, it's still early <1.0.0 days. The signature for patch* has also changed as a result.
We can make a new minor of kube when this is released.

Reply:
I could not convince myself that breakage is necessary: ‘kubectl’ is a command line tool, it has different purpose than “pure” api crate like that one, thus I guess it’s fine to have different default.